test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podsready
+
+import (
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
+	const (
+		instanceKey = "cloud.provider.com/instance"
+	)
+
+	var (
+		ns            *corev1.Namespace
+		defaultFlavor *kueue.ResourceFlavor
+	)
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+		defaultFlavor = testing.MakeResourceFlavor("default").Label(instanceKey, "default").Obj()
```

Comment:
To simplify the test:

```suggestion
		defaultFlavor = testing.MakeResourceFlavor("default").Obj()
```

Reply:
Also, you can create the RF once for the suite in `BeforeSuite`. We don't really need to be recreating it, making our tests cases a bit slower.

Reply:
Hm, I'm wondering about the new context instead of suite though, I have two concerns:
- the framework is currently setup in BeforeSuite, so it would require refactoring to move the configuration from `suite_test.go` to `scheduler_test.go`, then it no longer can be BeforeSuite but rather BeforeAll or BeforeEach.
- new context means that tests within the suite can no longer execute in parallel (do we support [Serial] tag in Kueue?), it may actually result in worse performance.

Reply:
Ah right... we share the same controllers across the suite.

Separate suite makes sense. I didn't know about `BeforeAll`, we should be using it in other tests. And we should create the common RF either `BeforeSuite` or `BeforeAll`. But that's a nit.

Reply:
Moved the RF setup to `BeforeSuite`, but it does not make any noticeable speedup on my local machine.
I think `BeforeAll` might be tricky as it works, but requires the container to be ordered, then all tests within the container are executed sequentially, which in our case might not be necessary.